WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt

```diff
@@ -230,6 +232,24 @@ class PostListMainViewModel @Inject constructor(
         }
     }
 
+    private fun getInitialAuthorFilterSelection(): AuthorFilterSelection {
+        return if (isFilteringByAuthorSupported()) {
+            prefs.postListAuthorSelection
+        } else {
+            AuthorFilterSelection.EVERYONE
+        }
+    }
+
+    /**
+     * Filtering by author is disable on:
+     * 1) Self-hosted sites - The XMLRPC api doesn't support filtering by author.
+     * 2) Self-hosted JetPack powered sites - we'd need support on the API for filtering by self-hosted user id.
```

Comment:
> we'd need support on the API for filtering by self-hosted user id.

Actually, this is exactly the opposite. We currently pass the wp.com user id to the API and it tries to filter it for the self-hosted id because not all users will have wp.com accounts. I don't think we should mention the API at all in this comment, because there is not a great solution in the API side either. (at least not without versioning)

My suggested comment would be:
```
2) Jetpack sites - we need to pass in the self-hosted user id to be able to filter for authors which we currently can't
```
or something along these lines.

Reply:
👍 Fixed https://github.com/wordpress-mobile/WordPress-Android/pull/9583/commits/97ddf6d9d1b34971ef834acce6632e43bcfd14ee